runtime/array.c

```diff
@@ -329,7 +331,8 @@ CAMLprim value caml_make_vect(value len, value init)
       /* We now know that [init] is not in the minor heap, so there is
          no need to call [caml_initialize]. */
       for (i = 0; i < size; i++) Field(res, i) = init;
-      res = caml_check_urgent_gc (res);
+      caml_check_urgent_gc (Val_unit);
+      caml_memprof_handle_postponed (Val_unit);
```

Comment:
In addition to using `caml_process_pending_actions` here instead of `caml_memprof_handle_postponed`, I would suggest that you do it even for small arrays. There is no reason not to do polling systematically (apart perhaps a small performance impact, but since `caml_process_pending_actions` is very fast, that should not be an issue).

Reply:
(And, of course, you can do this for other instances below.

Reply:
Done.